packages/flame_test/lib/src/flame_test.dart

```diff
@@ -19,13 +19,15 @@ extension FlameGameExtension on Component {
   Future<void> ensureAdd(Component component) async {
     await add(component);
     updateTree(0);
+    update(0);
```

Comment:
Future "Game in game" tests could be affected by this, since `update` then would be called twice. We`ll probably need a check of the parent here too.

Reply:
In theory, calling `update(0)` multiple times should not make a difference.

But in general, I was thinking that maybe `ensureAdd`/`ensureAddAll` methods can be moved into the `Component` class, in which case `FlameGame` would be able to redefine them to call `update` instead of `updateTree`.

Reply:
Hmm, don't think we'd want to add that directly in `Component` since this only should be used for tests.
But you are right, since two `0`s shouldn't make a difference we can just as well have it like this for now.

Reply:
Wait, but why is `update(0)` called here? This function only exist so that the component will be added directly to the tree, and that will happen already from `updateTree`?

Reply:
Because without it, when we call `flameGame.ensureAdd(component)` -- it would invoke `invokeTree`, which would in turn skip the `update()`. So, if there is a `MyFlameGame` that defines its own functionality in `update()`, then that functionality won't be invoked from `ensureAdd`.

Reply:
If the extension is specific to Component, then why is it called `FlameGameExtension`, and not `ComponentExtension`?

Reply:
Because we simply forgot to rename it when changing from having it specific for `FlameGame` to `Component`, it has `on Component` so can only be used on `Component`s. And even if it was specific for `FlameGame`, `update` wouldn't have had to be called here.